src/test/java/com/browserstack/automate/ci/jenkins/local/JenkinsBrowserStackLocalTest.java

```diff
@@ -0,0 +1,24 @@
+package com.browserstack.automate.ci.jenkins.local;
+
+import org.junit.runner.RunWith;
+import com.pholser.junit.quickcheck.Property;
+import com.pholser.junit.quickcheck.runner.JUnitQuickcheck;
+import static org.assertj.core.api.Assertions.assertThat;
+
+
+@RunWith(JUnitQuickcheck.class)
+public class JenkinsBrowserStackLocalTest {
+
+    public final String ACCESS_KEY = "dummy";
+
+    @Property
+    public void testLocalIdentifierGeneration(String buildTag) {
+        LocalConfig localConfig = new LocalConfig();
+        JenkinsBrowserStackLocal jenkinsBSLocal = new JenkinsBrowserStackLocal(ACCESS_KEY, localConfig, buildTag);
+
+        String localIdentifier = jenkinsBSLocal.getLocalIdentifier();
+
+        assertThat(localIdentifier).matches("^[a-zA-Z0-9_\\-\\.]+$");
```

Comment:
can you use stronger validation?

Reply:
What do you suggest?

Reply:
isEqualTo is stronger than matches

Reply:
Oh, I see. The reason I've used a match here rather than asserting equality is because the input is generated by the property-based testing library.

When I'm making the assertion I don't know exactly what the resulting string is going to look like, and I don't care. All that I care about is that it contains only valid characters.

Reply:
Sounds good